src/components/encounter/encounter-form.component.tsx

```diff
@@ -85,12 +80,18 @@ const EncounterForm: React.FC<EncounterFormProps> = ({
   setAllInitialValues,
   isSubmitting,
 }) => {
+  const formSessionDate = useMemo(() => new Date(), []);
   const { t } = useTranslation();
+  const session = useSession();
+  const { encounterRole: role } = useEncounterRole();
```

Comment:
We should update the [encounter context's dependencies loader](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/components/encounter/encounter-form.component.tsx#L127) accordingly to correctly initialise the role(s) related field(s)